src/cli/hdcanvas.rs

```diff
@@ -85,11 +85,21 @@ impl Canvas {
                     let p_bottom = self.pixel(2 * i_div_2 + 1, j);
 
                     match (p_top, p_bottom) {
-                        (Some(top), Some(bottom)) => write!(
-                            out,
-                            "{}",
-                            self.brush.paint("▀", top.ansi_style().on(bottom))
-                        )?,
+                        (Some(top), Some(bottom)) => {
+                            if top == bottom {
+                                write!(
+                                    out,
+                                    "{}",
+                                    self.brush.paint(" ", top.ansi_style().on(bottom))
+                                )?
+                            } else {
```

Comment:
what about the `else` path? Isn't that still potentially causing problems if kitty changes colors?

Reply:
Potentially, however I think it is only used in the checkerboard, which has a fairly high luminosity difference.  I think kitty uses a simple percentage luminosity difference (I want to work on the compare functionality from #21 next and would like to add the value that kitty uses).  I set it to 40 which seems high enough to avoid issues with text and the checkerboard still works.  Maybe a comment about this potential issue would be appropriate to avoid it being used for something else in the future.

Reply:
I just checked and 40 is the highest value of text_fg_override_threshold that the current checkerboard works with, so I guess that would be a potential reason to use the graphics protocol, although I'm not sure if anyone will want to set it higher than that.  Does look a bit odd when it triggers.